python/pyspark/serializers.py

```diff
@@ -267,13 +267,13 @@ def load_stream(self, stream):
         """
         Deserialize ArrowRecordBatches to an Arrow table and return as a list of pandas.Series.
         """
-        from pyspark.sql.types import _check_dataframe_localize_timestamps
+        from pyspark.sql.types import _check_series_localize_timestamps
         import pyarrow as pa
         reader = pa.open_stream(stream)
         for batch in reader:
             # NOTE: changed from pa.Columns.to_pandas, timezone issue in conversion fixed in 0.7.1
-            pdf = _check_dataframe_localize_timestamps(batch.to_pandas(), self._timezone)
-            yield [c for _, c in pdf.iteritems()]
+            yield [_check_series_localize_timestamps(c.to_pandas(), self._timezone)
+                   for c in pa.Table.from_batches([batch]).itercolumns()]
```

Comment:
Maybe we can remove the comment above (`# NOTE: ...`) ?

Reply:
I actually don't know what the comment above means. @BryanCutler do you remember?

Reply:
Yeah, the note was because prev we iterated over Arrow columns and converted each to a Series, then changed to convert an Arrow batch to DataFrame and then iterated over DataFrame columns to get a Series.  I wasn't sure if there might be a perf decrease, so I left the note but I'm not sure why it wasn't done like the above in the first place - seems like it would be just as good as the original.  Anyway, yeah the note can be removed now.

Reply:
@BryanCutler Thanks for the clarification. I removed the note.